test/arborist/build-ideal-tree.js

```diff
@@ -2392,3 +2406,65 @@ t.test('set the current on ERESOLVE triggered by devDeps', async t => {
     },
   })
 })
+
+t.test('shrinkwrapped dev/optional deps should not clobber flags', t => {
+  t.test('optional', async t => {
+    const path = t.testdir({
+      'package.json': JSON.stringify({
+        name: 'project',
+        version: '1.2.3',
+        optionalDependencies: {
+          '@isaacs/test-package-with-shrinkwrap': '^1.0.0',
+        },
+      }),
+    })
+    const tree = await buildIdeal(path, { complete: true })
+    const swName = '@isaacs/test-package-with-shrinkwrap'
+    const swDep = tree.children.get(swName)
+    const metaDep = swDep.children.get('abbrev')
+    t.equal(swDep.optional, true, 'shrinkwrapped dep is optional')
+    t.equal(metaDep.optional, true, 'shrinkwrapped metadep optional')
+
+    // make sure we're not just somehow leaving ALL flags true
+    t.equal(swDep.dev, false, 'sw dep is not dev')
+    t.equal(metaDep.dev, false, 'meta dep is not dev')
+  })
+
+  t.test('dev', async t => {
+    const path = t.testdir({
+      'package.json': JSON.stringify({
+        name: 'project',
+        version: '1.2.3',
+        devDependencies: {
+          '@isaacs/test-package-with-shrinkwrap': '^1.0.0',
+        },
+      }),
+    })
+    const tree = await buildIdeal(path, { complete: true })
+    const swName = '@isaacs/test-package-with-shrinkwrap'
+    const swDep = tree.children.get(swName)
+    const metaDep = swDep.children.get('abbrev')
+    t.equal(swDep.dev, true, 'shrinkwrapped dep is dev')
+    t.equal(metaDep.dev, true, 'shrinkwrapped metadep dev')
+
+    // make sure we're not just somehow leaving ALL flags true
+    t.equal(swDep.optional, false, 'sw dep is not optional')
+    t.equal(metaDep.optional, false, 'meta dep is not optional')
+  })
+
+  t.end()
+})
+
+t.test('do not ERESOLVE on peerOptionals that are ignored anyway', t => {
+  // this simulates three cases where a conflict occurs during the peerSet
+  // generation phase, but will not manifest in the tree building phase.
+  const base = resolve(fixtures, 'peer-optional-eresolve')
+  const cases = ['a', 'b', 'c']
+  t.plan(cases.length)
+  for (const c of cases) {
+    t.test(`case ${c}`, async t => {
+      const path = resolve(base, c)
+      t.matchSnapshot(await buildIdeal(path))
```

Comment:
```suggestion
      t.matchSnapshot(await printIdeal(path))
```

Reply:
ahhhh, right, because the print does the cleanup.  Thanks!